src/nexus-prisma/NexusPrismaBuilder.ts

```diff
@@ -64,32 +60,23 @@ export class NexusPrismaBuilder {
     }
   }
 
-  getNexusPrismaMethod() {
-    return [
-      this.getCRUDDynamicOutputMethod(),
-      this.getModelDynamicOutputMethod(),
-      ...this.getPrismaScalars(),
-    ]
+  build() {
+    return [this.buildCRUD(), this.buildModel(), ...this.buildScalers()]
   }
 
   /**
    * Generate `t.crud` output method
    */
-  protected getCRUDDynamicOutputMethod() {
+  protected buildCRUD(): DynamicOutputPropertyDef<'crud'> {
     //const methodName = this.params.methodName ? this.params.methodName : 'crud';
-
     return dynamicOutputProperty({
       name: 'crud',
       typeDefinition: `: NexusPrisma<TypeName, 'crud'>`,
       factory: ({ typeDef: t, typeName: graphQLTypeName }) => {
-        if (graphQLTypeName !== 'Query' && graphQLTypeName !== 'Mutation') {
-          throw new Error(
-            `t.crud can only be used on a 'Query' & 'Mutation' objectType. Please use 't.model' instead`,
-          )
-        }
+        let mappedFields: FieldsWithModelName[]
```

Comment:
Aside: Have not fully wrapped my head around `mappedFields`. I started reading the dmmf module tonight but still hazy.

Reply:
Instead of early return we do branch assignment and then continue working within this function.

Reply:
In the DMMF format, `mappings` is an array of object which maps for each model in the prisma schema, the type of operation to the corresponding graphql field name. By type of operation, I mean `findOne`, `findMany`, `create`, `update`, etc.
In other words, if you have a `Blog` model in your schema, mapping will look something like this:
```js
{
 "model": "Blog",
 "findOne": "blog",
 "findMany": "blogs",
 "create": "createBlog",
 "update": "updateBlog"
}
```
While it might seem like we can compute the name of these fields, the goal is here is to have one single source of truth (DMMF is computed by the core), especially regarding pluralization

As for `mappedFields`, it's just a small datastructure that combine the mappings of a type with the graphql field information for each of the field names given by the mapping

Reply:
> Instead of early return we do branch assignment and then continue working within this function

Same as above, that's a code-style that I started to use more and more.

- I generally prefer 3 separated if with a clear condition, rather than a if/elseif/elseif/else, which IMO makes it harder to read because of conditions that rely on the previous ones.

- You might argue that this pattern can lead to verbose conditions, and that's true. I use it when it's not the case, and generally, never use `else` because the ifs before already contain early return, so we can basically just replace the else by a simple return in the function scope.

- I also find early returns to make the code mentally easier, as you can digest the code block by block (because it stops), without having to take into account whatever will happen after

- In this case, it also prevents you from messing with the `mappedFields` variable by mistake (as we used to have two const instead of one let)

Finally, while this might seem a lot for that simple example, these are some arguments that mostly apply to more complex example, but that I ended up using even for simple cases out of habits

PS: I don't feel necessarily strong about keeping the way it was. I'll try to develop these kinds of habits/reasoning as we're starting our collaboration so that you have an idea of my programming style and that we slowly find a common ground to code with some norms/same patterns 🙌

Reply:
Thanks for sharing this!

I tried a few experiments;

#### if-guard + ternary + never-default

![image](https://user-images.githubusercontent.com/284476/64257253-0bb07800-cef3-11e9-9d0f-e5a9e41f43a8.png)

#### "pattern-matching" + throw-default

TS bug: throw default leads TS to see unreachable code

![image](https://user-images.githubusercontent.com/284476/64257346-37336280-cef3-11e9-8655-2c1efd7db528.png)

#### if-guard + "pattern-matching"

TS bug: variable is seen as unassigned unless never-default provided

![image](https://user-images.githubusercontent.com/284476/64257578-95f8dc00-cef3-11e9-9ed2-3bdc49635dfa.png)

Of these, I think `if-guard + ternary + never-default` is best option here. And I think that brings us [more] into alignment too! 🙌